x/bankplus/keeper/keeper_test.go

```diff
@@ -0,0 +1,192 @@
+package keeper
+
+import (
+	"github.com/line/lbm-sdk/simapp"
+	sdk "github.com/line/lbm-sdk/types"
+	authkeeper "github.com/line/lbm-sdk/x/auth/keeper"
+	authtypes "github.com/line/lbm-sdk/x/auth/types"
+	"testing"
+
+	"github.com/line/lbm-sdk/x/bank/types"
+	ocproto "github.com/line/ostracon/proto/ostracon/types"
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	initialPower = int64(100)
+	holder       = "holder"
+	blocker      = "blocker"
+)
+
+var (
+	holderAcc  = authtypes.NewEmptyModuleAccount(holder)
+	blockedAcc = authtypes.NewEmptyModuleAccount(blocker)
+	burnerAcc  = authtypes.NewEmptyModuleAccount(authtypes.Burner, authtypes.Burner)
+
+	initTokens = sdk.TokensFromConsensusPower(initialPower)
+	initCoins  = sdk.NewCoins(sdk.NewCoin(sdk.DefaultBondDenom, initTokens))
+)
+
+// nolint: interfacer
+func getCoinsByName(ctx sdk.Context, bk Keeper, ak types.AccountKeeper, moduleName string) sdk.Coins {
+	moduleAddress := ak.GetModuleAddress(moduleName)
+	macc := ak.GetAccount(ctx, moduleAddress)
+	if macc == nil {
+		return sdk.Coins(nil)
+	}
+
+	return bk.GetAllBalances(ctx, macc.GetAddress())
+}
+
+type IntegrationTestSuite struct {
+	suite.Suite
+
+	app         *simapp.SimApp
+	ctx         sdk.Context
+	queryClient types.QueryClient
+}
+
+func (suite *IntegrationTestSuite) TestSupply_SendCoins() {
+	app := simapp.Setup(false)
+	ctx := app.BaseApp.NewContext(false, ocproto.Header{Height: 1})
+	appCodec := app.AppCodec()
+
+	// add module accounts to supply keeper
+	maccPerms := simapp.GetMaccPerms()
+	maccPerms[holder] = nil
+	maccPerms[authtypes.Burner] = []string{authtypes.Burner}
+	maccPerms[authtypes.Minter] = []string{authtypes.Minter}
+
+	authKeeper := authkeeper.NewAccountKeeper(
+		appCodec, app.GetKey(types.StoreKey), app.GetSubspace(types.ModuleName),
+		authtypes.ProtoBaseAccount, maccPerms,
+	)
+	keeper := NewBaseKeeper(
+		appCodec, app.GetKey(types.StoreKey), authKeeper,
+		app.GetSubspace(types.ModuleName), make(map[string]bool),
+	)
+
+	baseAcc := authKeeper.NewAccountWithAddress(ctx, authtypes.NewModuleAddress("baseAcc"))
+	suite.Require().NoError(keeper.SetBalances(ctx, holderAcc.GetAddress(), initCoins))
+
+	keeper.SetSupply(ctx, types.NewSupply(initCoins))
+	authKeeper.SetModuleAccount(ctx, holderAcc)
+	authKeeper.SetModuleAccount(ctx, burnerAcc)
+	authKeeper.SetAccount(ctx, baseAcc)
+
+	suite.Require().Panics(func() {
+		keeper.SendCoinsFromModuleToModule(ctx, "", holderAcc.GetName(), initCoins) // nolint:errcheck
+	})
+
+	suite.Require().Panics(func() {
+		keeper.SendCoinsFromModuleToModule(ctx, authtypes.Burner, "", initCoins) // nolint:errcheck
+	})
+
+	suite.Require().Panics(func() {
+		keeper.SendCoinsFromModuleToAccount(ctx, "", baseAcc.GetAddress(), initCoins) // nolint:errcheck
+	})
+
+	suite.Require().Error(
```

Comment:
Yes, that's right. 
This is because it is the amount of `initCoins` added to `initCoins`. The amount is double of `initCoins`. So the balance is not enough.